### PR TITLE
Vulnerability alerts

### PR DESCRIPTION
**Pull request by palvaroni**
At the moment vulnerabilityAlert.json doesn't probably match the actual payload, but is a hybrid between `create` and `dismiss` actions.

### CODE REVIEW

**Comment by savpek** [comment]
```
{
    "action": "create",
    "alert": {
        "id": 70620783,
        "affected_range": "< 3.0.2",
        "affected_package_name": "minimatch",
        "external_reference": "https://nvd.nist.gov/vuln/detail/CVE-2016-10540",
        "external_identifier": "CVE-2016-10540",
        "fixed_in": "3.0.2"
    },
    "repository": {
        "id": 160878320,
        "node_id": "MDEwOlJlcG9zaXRvcnkxNjA4NzgzMjA=",
        "name": "BCRepo3",
        "full_name": "octo-business-cloud-testing/BCRepo3",
        "private": false,
        "html_url": "https://github.com/octo-business-cloud-testing/BCRepo3",
        "description": "",
        "fork": false,
        "url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3",
        "forks_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/forks",
        "keys_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/keys{/key_id}",
        "collaborators_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/collaborators{/collaborator}",
        "teams_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/teams",
        "hooks_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/hooks",
        "issue_events_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/issues/events{/number}",
        "events_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/events",
        "assignees_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/assignees{/user}",
        "branches_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/branches{/branch}",
        "tags_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/tags",
        "blobs_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/blobs{/sha}",
        "git_tags_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/tags{/sha}",
        "git_refs_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/refs{/sha}",
        "trees_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/trees{/sha}",
        "statuses_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/statuses/{sha}",
        "languages_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/languages",
        "stargazers_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/stargazers",
        "contributors_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/contributors",
        "subscribers_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/subscribers",
        "subscription_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/subscription",
        "commits_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/commits{/sha}",
        "git_commits_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/commits{/sha}",
        "comments_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/comments{/number}",
        "issue_comment_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/issues/comments{/number}",
        "contents_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/contents/{+path}",
        "compare_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/compare/{base}...{head}",
        "merges_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/merges",
        "archive_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/{archive_format}{/ref}",
        "downloads_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/downloads",
        "issues_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/issues{/number}",
        "pulls_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/pulls{/number}",
        "milestones_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/milestones{/number}",
        "notifications_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/notifications{?since,all,participating}",
        "labels_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/labels{/name}",
        "releases_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/releases{/id}",
        "deployments_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/deployments",
        "created_at": "2018-12-07T21:42:00Z",
        "updated_at": "2018-12-13T00:01:09Z",
        "pushed_at": "2018-12-13T00:01:07Z",
        "git_url": "git://github.com/octo-business-cloud-testing/BCRepo3.git",
        "ssh_url": "git@github.com:octo-business-cloud-testing/BCRepo3.git",
        "clone_url": "https://github.com/octo-business-cloud-testing/BCRepo3.git",
        "svn_url": "https://github.com/octo-business-cloud-testing/BCRepo3",
        "size": 0,
        "stargazers_count": 0,
        "watchers_count": 0,
        "has_issues": true,
        "has_projects": true,
        "has_downloads": true,
        "has_wiki": true,
        "has_pages": false,
        "forks_count": 0,
        "archived": false,
        "open_issues_count": 0,
        "forks": 0,
        "open_issues": 0,
        "watchers": 0,
        "default_branch": "master"
    },
    "organization": {
        "login": "octo-business-cloud-testing",
        "id": 43544715,
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjQzNTQ0NzE1",
        "url": "https://api.github.com/orgs/octo-business-cloud-testing",
        "repos_url": "https://api.github.com/orgs/octo-business-cloud-testing/repos",
        "events_url": "https://api.github.com/orgs/octo-business-cloud-testing/events",
        "hooks_url": "https://api.github.com/orgs/octo-business-cloud-testing/hooks",
        "issues_url": "https://api.github.com/orgs/octo-business-cloud-testing/issues",
        "members_url": "https://api.github.com/orgs/octo-business-cloud-testing/members{/member}",
        "public_members_url": "https://api.github.com/orgs/octo-business-cloud-testing/public_members{/member}",
        "avatar_url": "https://avatars3.githubusercontent.com/u/43544715?v=4",
    },
    "sender": {
        "login": "github",
        "id": 9919,
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjk5MTk=",
        "avatar_url": "https://avatars1.githubusercontent.com/u/9919?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/github",
        "html_url": "https://github.com/github",
        "followers_url": "https://api.github.com/users/github/followers",
        "following_url": "https://api.github.com/users/github/following{/other_user}",
        "gists_url": "https://api.github.com/users/github/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/github/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/github/subscriptions",
        "organizations_url": "https://api.github.com/users/github/orgs",
        "repos_url": "https://api.github.com/users/github/repos",
        "events_url": "https://api.github.com/users/github/events{/privacy}",
        "received_events_url": "https://api.github.com/users/github/received_events",
        "type": "Organization",
        "site_admin": false
    }
}
```

**Comment by savpek** [comment]
```
{
    "action": "resolve",
    "alert": {
        "id": 70619571,
        "affected_range": "< 4.2.1",
        "affected_package_name": "hoek",
        "external_reference": "https://nvd.nist.gov/vuln/detail/CVE-2018-3728",
        "external_identifier": "CVE-2018-3728",
        "fixed_in": "4.2.1"
    },
    "repository": {
        "id": 160878320,
        "node_id": "MDEwOlJlcG9zaXRvcnkxNjA4NzgzMjA=",
        "name": "BCRepo3",
        "full_name": "octo-business-cloud-testing/BCRepo3",
        "private": false,
        "html_url": "https://github.com/octo-business-cloud-testing/BCRepo3",
        "description": "",
        "fork": false,
        "url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3",
        "forks_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/forks",
        "keys_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/keys{/key_id}",
        "collaborators_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/collaborators{/collaborator}",
        "teams_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/teams",
        "hooks_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/hooks",
        "issue_events_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/issues/events{/number}",
        "events_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/events",
        "assignees_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/assignees{/user}",
        "branches_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/branches{/branch}",
        "tags_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/tags",
        "blobs_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/blobs{/sha}",
        "git_tags_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/tags{/sha}",
        "git_refs_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/refs{/sha}",
        "trees_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/trees{/sha}",
        "statuses_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/statuses/{sha}",
        "languages_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/languages",
        "stargazers_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/stargazers",
        "contributors_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/contributors",
        "subscribers_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/subscribers",
        "subscription_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/subscription",
        "commits_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/commits{/sha}",
        "git_commits_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/git/commits{/sha}",
        "comments_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/comments{/number}",
        "issue_comment_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/issues/comments{/number}",
        "contents_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/contents/{+path}",
        "compare_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/compare/{base}...{head}",
        "merges_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/merges",
        "archive_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/{archive_format}{/ref}",
        "downloads_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/downloads",
        "issues_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/issues{/number}",
        "pulls_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/pulls{/number}",
        "milestones_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/milestones{/number}",
        "notifications_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/notifications{?since,all,participating}",
        "labels_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/labels{/name}",
        "releases_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/releases{/id}",
        "deployments_url": "https://api.github.com/repos/octo-business-cloud-testing/BCRepo3/deployments",
        "created_at": "2018-12-07T21:42:00Z",
        "updated_at": "2018-12-12T23:59:45Z",
        "pushed_at": "2018-12-13T00:00:21Z",
        "git_url": "git://github.com/octo-business-cloud-testing/BCRepo3.git",
        "ssh_url": "git@github.com:octo-business-cloud-testing/BCRepo3.git",
        "clone_url": "https://github.com/octo-business-cloud-testing/BCRepo3.git",
        "svn_url": "https://github.com/octo-business-cloud-testing/BCRepo3",
        "homepage": "",
        "size": 0,
        "stargazers_count": 0,
        "watchers_count": 0,
        "language": "",
        "has_issues": true,
        "has_projects": true,
        "has_downloads": true,
        "has_wiki": true,
        "has_pages": false,
        "forks_count": 0,
        "mirror_url": "",
        "archived": false,
        "open_issues_count": 0,
        "license": "",
        "forks": 0,
        "open_issues": 0,
        "watchers": 0,
        "default_branch": "master"
    },
    "organization": {
        "login": "octo-business-cloud-testing",
        "id": 43544715,
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjQzNTQ0NzE1",
        "url": "https://api.github.com/orgs/octo-business-cloud-testing",
        "repos_url": "https://api.github.com/orgs/octo-business-cloud-testing/repos",
        "events_url": "https://api.github.com/orgs/octo-business-cloud-testing/events",
        "hooks_url": "https://api.github.com/orgs/octo-business-cloud-testing/hooks",
        "issues_url": "https://api.github.com/orgs/octo-business-cloud-testing/issues",
        "members_url": "https://api.github.com/orgs/octo-business-cloud-testing/members{/member}",
        "public_members_url": "https://api.github.com/orgs/octo-business-cloud-testing/public_members{/member}",
        "avatar_url": "https://avatars3.githubusercontent.com/u/43544715?v=4",
        "description": ""
    },
    "sender": {
        "login": "github",
        "id": 9919,
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjk5MTk=",
        "avatar_url": "https://avatars1.githubusercontent.com/u/9919?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/github",
        "html_url": "https://github.com/github",
        "followers_url": "https://api.github.com/users/github/followers",
        "following_url": "https://api.github.com/users/github/following{/other_user}",
        "gists_url": "https://api.github.com/users/github/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/github/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/github/subscriptions",
        "organizations_url": "https://api.github.com/users/github/orgs",
        "repos_url": "https://api.github.com/users/github/repos",
        "events_url": "https://api.github.com/users/github/events{/privacy}",
        "received_events_url": "https://api.github.com/users/github/received_events",
        "type": "Organization",
        "site_admin": false
    }
}
```

File: Tests/RealDataTests.cs [hunk 1]
```diff
@@ -149,5 +149,41 @@ public void WhenJenkinsBuildFails_ThenSendMessage()
...
+                    slack.Received(1).Send(Arg.Is<string>("#general"), Arg.Any<SlackMessageModel>()));
+        }
+
+        [Fact]
+        public void WhenVulnerabilityAlertIsDismissed_ThenSendMessage()
+        {
+            ActionTestBuilder<VulnerabilityAlertAction>
+                .Create((slack, logger) => new VulnerabilityAlertAction(slack, logger))
+                .ExecuteWith("vulnerabilityAlert.dismiss.json", slackChannels: "#general")
+                .AssertInvokedOn(requestType: "repository_vulnerability_alert", requestAction: "create")
```
Comment (palvaroni) [inline Tests/RealDataTests.cs]: requestAction should be "dismiss"?

File: Tests/RealDataTests.cs [hunk 2]
```diff
@@ -149,5 +149,41 @@ public void WhenJenkinsBuildFails_ThenSendMessage()
...
+                    slack.Received(1).Send(Arg.Is<string>("#general"), Arg.Any<SlackMessageModel>()));
+        }
+
+        [Fact]
+        public void WhenVulnerabilityAlertIsResolved_ThenSendMessage()
+        {
+            ActionTestBuilder<VulnerabilityAlertAction>
+                .Create((slack, logger) => new VulnerabilityAlertAction(slack, logger))
+                .ExecuteWith("vulnerabilityAlert.resolve.json", slackChannels: "#general")
+                .AssertInvokedOn(requestType: "repository_vulnerability_alert", requestAction: "create")
```
Comment (palvaroni) [inline Tests/RealDataTests.cs]: requestAction should be "resolve"?

File: README.md [hunk 3]
```diff
@@ -58,9 +66,14 @@ App supports globally configured actions which are invoked on every repository e
 | issue_label                 | Notifies channel about labels on issues. | Supports filtering with `data: [ "needs help", "bug" ]`, if data is not defined all labels are accepted. |
 | pullrequest_label           | Notifies channel about new labels on pull requests. | Supports filtering with `data: [ "needs help", "bug" ]`, if data is not defined all labels are accepted. |
 | jenkins_build_error         | Jenkins ci build failed. |  |
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (savpek) [reply README.md]: fuuuuu 😢

**Pull request merged**